sqsconsumer.go

```diff
@@ -178,50 +182,71 @@ func (m *SmartSQSConsumer) StartConsuming(ctx context.Context) error {
 
 // worker function represents a single "message worker." worker will infinitely process messages until
 // messages is closed. worker is responsible for handling deletion of messages, or handling
-// messages that have retryable error.
+// messages that have retryable error. On the event of a retryable error, worker will change
+// the visibilitytimeout of a message to be the configured retryableErr.VisibilityTimeout
 func (m *SmartSQSConsumer) worker(ctx context.Context, messages <-chan *sqs.Message) {
 	for message := range messages {
 		err := m.GetSQSMessageConsumer().ConsumeMessage(ctx, message)
 		if err != nil {
 			switch err.(type) {
 			case RetryableConsumerError:
+				// check to see if we've reached the maximum number of retries allowed
+				receiveCount := getApproximateReceiveCount(message)
+				if receiveCount > m.MaxRetries {
+					m.ackMessage(ctx, func() error {
+						return m.deleteMessage(message)
```

Comment:
task: can you please emit the `sqs.consumer_error.retries_exceeded` stat here?

Reply:
i’ve updated the PR with some new abstraction. This PR does not address stat-ing retries like you wanted. However, I’d like to build in stat-ing retry instances in https://github.com/asecurityteam/sqsmessageconsumer-stat using the new interface this PR provides
Later in the future when we work on ticket 1093, i’d like to build in callback logic for permanent “Fallback” logic into runsqs, where we can then stat permanent retries.
are you cool with this? I really don’t like the idea of stats being thrown into a `MessageConsumer` implementation, as it then feels like we added some of the stats in a MessageConsumer , and then added the rest of the stats in sqsmessageconsumer-stat